src/view/utils/common.js

```diff
@@ -7,4 +9,16 @@ const getRandomInteger = (a = 0, b = 1) => {
 
 const getRandomIndex = (array) => getRandomInteger(0, array.length - 1);
 
-export { getRandomInteger, getRandomIndex };
+const isEscEvent = (evt) => evt.key === KeyboardKey.ESCAPE;
+
+const updateItem = (items, update) => {
+  const index = items.findIndex((currentValue) => currentValue.id === update.id);
+
+  if (index === -1) {
+    return items;
+  }
+
+  return [...items.slice(0, index), update, ...items.slice(index + 1)];
```

Comment:
Это что-то сложное, и в реальных проектах так не пишут

```suggestion
  return items.map((it) => it.id === update.id ? update : it)
```